# Desafío entregable #1: Elección de datasets potenciales

In [1]:
import pandas as pd

## Database #1: Spotify tracks dataset

#### Fuente: https://www.kaggle.com/datasets/maharshipandya/-spotify-tracks-dataset

In [2]:
df_s = pd.read_csv('Spotify_Track_Dataset.csv')

In [3]:
df_s.head()

Unnamed: 0                track_id                 artists  \
0           0  5SuOikwiRyPMVoIQDJUgSV             Gen Hoshino   
1           1  4qPNDBW1i3p13qLCt0Ki3A            Ben Woodward   
2           2  1iJBSr7s7jYXzM8EGcbK5b  Ingrid Michaelson;ZAYN   
3           3  6lfxq3CG4xtTiEg7opyCyx            Kina Grannis   
4           4  5vjLSffimiIP26QG5WcN2K        Chord Overstreet   

                                          album_name  \
0                                             Comedy   
1                                   Ghost (Acoustic)   
2                                     To Begin Again   
3  Crazy Rich Asians (Original Motion Picture Sou...   
4                                            Hold On   

                   track_name  popularity  duration_ms  explicit  \
0                      Comedy          73       230666     False   
1            Ghost - Acoustic          55       149610     False   
2              To Begin Again          57       210826     False   
3  Can't Help Falling In Love          71       201933     False   
4                     Hold On          82       198853     False   

   danceability  energy  ...  loudness  mode  speechiness  acousticness  \
0         0.676  0.4610  ...    -6.746     0       0.1430        0.0322   
1         0.420  0.1660  ...   -17.235     1       0.0763        0.9240   
2         0.438  0.3590  ...    -9.734     1       0.0557        0.2100   
3         0.266  0.0596  ...   -18.515     1       0.0363        0.9050   
4         0.618  0.4430  ...    -9.681     1       0.0526        0.4690   

   instrumentalness  liveness  valence    tempo  time_signature  track_genre  
0          0.000001    0.3580    0.715   87.917               4     acoustic  
1          0.000006    0.1010    0.267   77.489               4     acoustic  
2          0.000000    0.1170    0.120   76.332               4     acoustic  
3          0.000071    0.1320    0.143  181.740               3     acoustic  
4          0.000000    0.0829    0.167  119.949               4     acoustic  

[5 rows x 21 columns]

In [4]:
df_s.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114000 entries, 0 to 113999
Data columns (total 21 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Unnamed: 0        114000 non-null  int64  
 1   track_id          114000 non-null  object 
 2   artists           113999 non-null  object 
 3   album_name        113999 non-null  object 
 4   track_name        113999 non-null  object 
 5   popularity        114000 non-null  int64  
 6   duration_ms       114000 non-null  int64  
 7   explicit          114000 non-null  bool   
 8   danceability      114000 non-null  float64
 9   energy            114000 non-null  float64
 10  key               114000 non-null  int64  
 11  loudness          114000 non-null  float64
 12  mode              114000 non-null  int64  
 13  speechiness       114000 non-null  float64
 14  acousticness      114000 non-null  float64
 15  instrumentalness  114000 non-null  float64
 16  liveness          11

#### El archivo está compuesto por 114.000 canciones incluidas en el servicio Spotify, la información de cada una de esas canciones se encuentra distribuida en 21 columnas. Estas detallan el nombre de las canciones, los interpretes, el album del que forma parte, su duración, etcétera. El archivo también incluye columnas tales como "danceability", "energy", "speechiness", "acousticness", entre otros, que dotan de valores numéricos (generalmente entre 0 y 1) a dichos parámetros categóricos y subjetivos (más información en el link). Estas variables son útiles para caracterizar a grandes rasgos el enorme número de canciones disponibles en Spotify, con lo cual podríamos desarrollar un sistema de agrupamiento de canciones a partir de sus características en común. Esto permitiría desarrollar un sistema de calificación y recomendación de canciones a los usuarios en función de sus gustos.

In [5]:
round(df_s.describe(), 2)

Unnamed: 0  popularity  duration_ms  danceability     energy  \
count   114000.00   114000.00    114000.00     114000.00  114000.00   
mean     56999.50       33.24    228029.15          0.57       0.64   
std      32909.11       22.31    107297.71          0.17       0.25   
min          0.00        0.00         0.00          0.00       0.00   
25%      28499.75       17.00    174066.00          0.46       0.47   
50%      56999.50       35.00    212906.00          0.58       0.68   
75%      85499.25       50.00    261506.00          0.70       0.85   
max     113999.00      100.00   5237295.00          0.98       1.00   

             key   loudness       mode  speechiness  acousticness  \
count  114000.00  114000.00  114000.00    114000.00     114000.00   
mean        5.31      -8.26       0.64         0.08          0.31   
std         3.56       5.03       0.48         0.11          0.33   
min         0.00     -49.53       0.00         0.00          0.00   
25%         2.00     -10.01       0.00         0.04          0.02   
50%         5.00      -7.00       1.00         0.05          0.17   
75%         8.00      -5.00       1.00         0.08          0.60   
max        11.00       4.53       1.00         0.96          1.00   

       instrumentalness   liveness    valence      tempo  time_signature  
count         114000.00  114000.00  114000.00  114000.00       114000.00  
mean               0.16       0.21       0.47     122.15            3.90  
std                0.31       0.19       0.26      29.98            0.43  
min                0.00       0.00       0.00       0.00            0.00  
25%                0.00       0.10       0.26      99.22            4.00  
50%                0.00       0.13       0.46     122.02            4.00  
75%                0.05       0.27       0.68     140.07            4.00  
max                1.00       1.00       1.00     243.37            5.00

In [6]:
df_s.isna().sum()

Unnamed: 0          0
track_id            0
artists             1
album_name          1
track_name          1
popularity          0
duration_ms         0
explicit            0
danceability        0
energy              0
key                 0
loudness            0
mode                0
speechiness         0
acousticness        0
instrumentalness    0
liveness            0
valence             0
tempo               0
time_signature      0
track_genre         0
dtype: int64

#### En la celda inmediatamente anterior se verificó el número de entradas vacías o NaN en cada columna, podemos observar que sólo tres columnas poseen sólo un valor de NaNs. Esto es sumamente positivo, indicando que muy probablemente el dataset se encuentre completo.  

## Database #2: Air traffic passenger statistics

#### Fuente: https://www.kaggle.com/datasets/thedevastator/airlines-traffic-passenger-statistics

In [7]:
df_at = pd.read_csv('Air_Traffic_Passenger_Statistics.csv')

In [8]:
df_at.head()

index  Activity Period Operating Airline Operating Airline IATA Code  \
0      0           200507      ATA Airlines                          TZ   
1      1           200507      ATA Airlines                          TZ   
2      2           200507      ATA Airlines                          TZ   
3      3           200507       Air Canada                           AC   
4      4           200507       Air Canada                           AC   

  Published Airline Published Airline IATA Code    GEO Summary GEO Region  \
0      ATA Airlines                          TZ       Domestic         US   
1      ATA Airlines                          TZ       Domestic         US   
2      ATA Airlines                          TZ       Domestic         US   
3       Air Canada                           AC  International     Canada   
4       Air Canada                           AC  International     Canada   

  Activity Type Code Price Category Code    Terminal Boarding Area  \
0           Deplaned            Low Fare  Terminal 1             B   
1           Enplaned            Low Fare  Terminal 1             B   
2     Thru / Transit            Low Fare  Terminal 1             B   
3           Deplaned               Other  Terminal 1             B   
4           Enplaned               Other  Terminal 1             B   

   Passenger Count Adjusted Activity Type Code  Adjusted Passenger Count  \
0            27271                    Deplaned                     27271   
1            29131                    Enplaned                     29131   
2             5415          Thru / Transit * 2                     10830   
3            35156                    Deplaned                     35156   
4            34090                    Enplaned                     34090   

   Year Month  
0  2005  July  
1  2005  July  
2  2005  July  
3  2005  July  
4  2005  July

In [9]:
df_at.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15007 entries, 0 to 15006
Data columns (total 17 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   index                        15007 non-null  int64 
 1   Activity Period              15007 non-null  int64 
 2   Operating Airline            15007 non-null  object
 3   Operating Airline IATA Code  14953 non-null  object
 4   Published Airline            15007 non-null  object
 5   Published Airline IATA Code  14953 non-null  object
 6   GEO Summary                  15007 non-null  object
 7   GEO Region                   15007 non-null  object
 8   Activity Type Code           15007 non-null  object
 9   Price Category Code          15007 non-null  object
 10  Terminal                     15007 non-null  object
 11  Boarding Area                15007 non-null  object
 12  Passenger Count              15007 non-null  int64 
 13  Adjusted Activity Type Code  15

#### Este dataset contiene información sobre el tráfico aéreo de transporte de pasajeros del 2017. Este brinda información sobre los distintos vuelos (15007 entradas), tales como nombre de la aerolínea, fecha del vuelo, número de pasajeros, origen y destino del vuelo, precio del vuelo, entre otras columnas (17 columnas en total). Estos datos podrían ser usados para estudiar el patrón de vuelos realizados en ese año ser usados para predecir los vuelos en años futuros. También se podría analizar el efecto de variables tales como el momento del año, el precio del vuelo, entre otras variables.

In [10]:
round(df_at.describe(), 2)

index  Activity Period  Passenger Count  Adjusted Passenger Count  \
count  15007.00         15007.00         15007.00                  15007.00   
mean    7503.00        201045.07         29240.52                  29331.92   
std     4332.29           313.34         58319.51                  58284.18   
min        0.00        200507.00             1.00                      1.00   
25%     3751.50        200803.00          5373.50                   5495.50   
50%     7503.00        201011.00          9210.00                   9354.00   
75%    11254.50        201308.00         21158.50                  21182.00   
max    15006.00        201603.00        659837.00                 659837.00   

           Year  
count  15007.00  
mean    2010.39  
std        3.14  
min     2005.00  
25%     2008.00  
50%     2010.00  
75%     2013.00  
max     2016.00

In [11]:
df_at.isna().sum()

index                           0
Activity Period                 0
Operating Airline               0
Operating Airline IATA Code    54
Published Airline               0
Published Airline IATA Code    54
GEO Summary                     0
GEO Region                      0
Activity Type Code              0
Price Category Code             0
Terminal                        0
Boarding Area                   0
Passenger Count                 0
Adjusted Activity Type Code     0
Adjusted Passenger Count        0
Year                            0
Month                           0
dtype: int64

#### Se verifica la presencia de valores NaN en el dataset, encontrándolos sólo en las columnas Airline IATA Code. Estas podrían ser posiblemente subsanadas con facilidad, puesto que los código IATA son claves constituidas por dos caracteres alfanuméricos, con las cuales se identifican las aerolíneas (https://en.wikipedia.org/wiki/Airline_codes).

## Database #3: Video game sales

#### Fuente: https://data.world/sumitrock/video-games-sales

In [12]:
df_vg = pd.read_csv('Video_Games.csv')

In [13]:
df_vg.head()

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  
0          51.0          8       322.0  Nintendo      E  
1           NaN        NaN         NaN       NaN    NaN  
2          73.0        8.3       709.0  Nintendo      E  
3          73.0          8       192.0  Nintendo      E  
4           NaN        NaN         NaN       NaN    NaN

In [14]:
df_vg.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       10015 non-null  object 
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  object 
 15  Rating           9950 non-null   object 
dtypes: float64(9), object(7)
memory usage: 2.0+ MB


#### Finalmente, la database de venta de video juegos está constituida por la información de 16719 títulos lanzados en diferentes consolas de videojuegos. La información de cada título está distribuida en 16 columnas, listando el número de los videojuegos, su consola de lanzamiento, el año de publicación, su género, ventas en EEUU, Europa y Japón, así como puntuaciones de usuarios y prensa especializada, etcétera. Estos datos podrían ser utilizados para determinar en que consola las ventas son superiores, que género rinde mejor en cada consola y en que región las ventas son las más altas, entre otras evaluaciones.

In [15]:
round(df_vg.describe(), 2)

Year_of_Release  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
count         16450.00  16719.00  16719.00  16719.00     16719.00   
mean           2006.49      0.26      0.15      0.08         0.05   
std               5.88      0.81      0.50      0.31         0.19   
min            1980.00      0.00      0.00      0.00         0.00   
25%            2003.00      0.00      0.00      0.00         0.00   
50%            2007.00      0.08      0.02      0.00         0.01   
75%            2010.00      0.24      0.11      0.04         0.03   
max            2020.00     41.36     28.96     10.22        10.57   

       Global_Sales  Critic_Score  Critic_Count  User_Count  
count      16719.00       8137.00       8137.00     7590.00  
mean           0.53         68.97         26.36      162.23  
std            1.55         13.94         18.98      561.28  
min            0.01         13.00          3.00        4.00  
25%            0.06         60.00         12.00       10.00  
50%            0.17         71.00         21.00       24.00  
75%            0.47         79.00         36.00       81.00  
max           82.53         98.00        113.00    10665.00

In [16]:
df_vg.isna().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         6704
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

#### A diferencia de los dos dataset anteriores, este muestra varias columnas con datos ausentes y numerosas entradas a las que les faltan datos, esto podría limitar el alcance del análisis de los datos.